src/util.ts

```diff
@@ -891,6 +891,43 @@ export class Util {
         };
     }
 
+    /**
+     * Given a list of ranges, create a range that starts with the first non-null lefthand range, and ends with the first non-null
+     * righthand range. Returns undefined if none of the items have a range.
+     */
+    public createBoundingRange(...locatables: Array<{ range?: Range }>) {
```

Comment:
Very welcome and should be applied to a bunch of existing statements

Reply:
Agreed. This has been in the `v1` branch for a while now, and most of the AST nodes were refactored to use this function there. I should probably do something similar in master at some point since that is a non-breaking change.